sklearn/utils/sparsefuncs_fast.pyx

```diff
@@ -67,24 +67,38 @@ def csr_mean_variance_axis0(X):
         Feature-wise variances
 
     """
-    cdef unsigned int n_samples = X.shape[0]
-    cdef unsigned int n_features = X.shape[1]
-
-    cdef np.ndarray[DOUBLE, ndim=1, mode="c"] X_data
-    X_data = np.asarray(X.data, dtype=np.float64)     # might copy!
-    cdef np.ndarray[int, ndim=1] X_indices = X.indices
+    if X.dtype == np.int32:
```

Comment:
By doing so, it is already enough to handle original related issues such as #5464 and #5776.